SalesforceCommerceCloud/pwa-kit
# Hooks to get the current page's Request and Response objects

To have more of a feature parity with `getProps`, we would like to make the current page's `Request` and `Response` objects to be available via hooks too. This PR simply make them accessible throughout the app by relying on React Context.

So the implementation is straightforward actually (basically I mostly copied what's already in #693). But what I'd like us to focus on is the experience of using the hooks, especially how it differs when they're inside `getProps` vs hooks.

Do the page request and response translate well to React hooks? 🤔 

Ticket: W-11668738

**Todos:**
- [ ] Revert my example code

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- React Context and hooks to share the `Request` and `Response` objects of the current page
- Some example of using the hooks in the template-typescript-minimal 

# How to Test-Drive This PR

```
cd packages/template-typescript-minimal
npm start
```

And then verify in the browser inspector (Network tab):
- the status code of the page response is `201` (now `202`)
- and it has the header `Some-Header: FOOBAR`

![Google Chrome 2022-08-31 at 16 13 45](https://user-images.githubusercontent.com/847300/187800999-8ca14056-0590-4829-884b-5a960ac9506e.png)


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
